### ISSUE TITLE
The `ivadomed` loss function is limited to only two arguments. Its compatibility requires an enhancement. 

### ISSUE THREAD

**Issue by rezazad68** [ivadomed/ivadomed#792]
If I understand right, `target_and_weights` is a list because right now, `loss_fct` must be called with exactly 2 args:

https://github.com/ivadomed/ivadomed/blob/a502b37bd988ea0eae2fd5fa002666a2003a16bb/ivadomed/training.py#L187

So you had to find a way to fit in a third arg, right? 

The problem with this is: On the surface, `JointsMSELoss` looks like it's compatible with any model, but I think it's actually only compatible with `HourglassNet`, because the loss function has a "secret" third argument.

_Originally posted by @joshuacwnewton in https://github.com/ivadomed/ivadomed/pull/787#discussion_r632496740_

**Comment by ahill187** [comment]
@rezazad68 this is the function you want to integrate, right? https://github.com/bearpaw/pytorch-pose/blob/master/pose/losses/jointsmseloss.py
